### PR TITLE
BTFS-1658: make happy path work for offline signing

### PR DESCRIPTION
**Pull request by steveyeom**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior?** (You can also refer to a JIRA ticket here)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)

* **Description of changes**

---

* **Please check if the PR fulfills these requirements**
- [ ] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [ ] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [ ] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [ x] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [ x] Code changes have run through `go fmt`
- [ x] Code changes have run through `go mod tidy`
- [ ] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

**Comment by steveyeom** [comment]
This PR makes offline signing based btfs upload work along with online signing.

File: core/commands/storage_offline.go [hunk 1]
```diff
@@ -6,19 +6,17 @@ import (
...
 	"github.com/TRON-US/go-btfs/core"
 	"github.com/TRON-US/go-btfs/core/commands/cmdenv"
 	"github.com/TRON-US/go-btfs/core/commands/storage"
 	"github.com/TRON-US/go-btfs/core/escrow"
-	"github.com/TRON-US/go-btfs/core/guard"
+	guard "github.com/TRON-US/go-btfs/core/guard"
 	escrowpb "github.com/tron-us/go-btfs-common/protos/escrow"
 	guardpb "github.com/tron-us/go-btfs-common/protos/guard"
 	ledgerpb "github.com/tron-us/go-btfs-common/protos/ledger"
+	"strconv"
```
Comment (taiyangc) [inline core/commands/storage_offline.go]: wrong line

File: core/commands/storage.go [hunk 2]
```diff
@@ -995,7 +998,7 @@ var storageUploadRecvContractCmd = &cmds.Command{
 func payWithSigning(ctx context.Context, req *cmds.Request, cfg *config.Config, n *core.IpfsNode, env cmds.Environment,
 	ss *storage.FileContracts, contractRequest *escrowpb.EscrowContractRequest) error {
 	// collecting all signed contracts means init status finished
-	ss.SendSessionStatusChan(storage.InitStatus, true, nil)
+	ss.SendSessionStatusChanPerMode(storage.InitStatus, true, nil, ctx)
```
Comment (taiyangc) [inline core/commands/storage.go]: I don't feel comfortable passing `ctx` around. Couldn't you try to set it on a `ss` field for this particular state?

File: core/commands/storage/session.go [hunk 3]
```diff
@@ -875,13 +893,37 @@ func (ss *FileContracts) UpdateSessionStatus(status int, succeed bool, err error
...
 
 func (ss *FileContracts) SendSessionStatusChan(status int, succeed bool, err error) {
+	ss.sendSessionStatusChan(status, succeed, err, false, context.TODO())
+}
+
+func (ss *FileContracts) SendSessionStatusChanPerMode(status int, succeed bool, err error, ctx context.Context) {
+	if ss.RunMode == OfflineSignMode {
+		ss.sendSessionStatusChan(status, succeed, err, true, ctx)
+	} else {
+		ss.sendSessionStatusChan(status, succeed, err, false, context.TODO())
```
Comment (taiyangc) [inline core/commands/storage/session.go]: There shouldn't any more `context.Background()` or `context.TODO()`, those are prone to memory leaks, consider always using `NewGoContext(req.Context)` or something similar.

File: core/commands/storage.go (hunk 2 above)
Comment (steveyeom) [reply core/commands/storage.go]: fixed

**Comment by steveyeom** [comment]
Reflected on the review comments.

**Comment by steveyeom** [comment]
Some more comments.

**Pull request merged**